### DISCUSSION TITLE
How to search the web using Google ?

### DISCUSSION THREAD

**Discussion by Dihelson** [open-webui/open-webui#3969] [Q&A]
I tried to follow the tutorial about getting a Google search API, but I'm stuck. I went to the PSE - Google programmable search, and created a new engine. But I could not find any API after that, as the tutorial suggests. The only way to get an API for what I understood, is through a paid mode on Google Cloud. The search engine I created gives me only a few lines of code to inserts on an existing website, no API. What to do ? What's the best solution to get the models search the web ? Thanks in advance. 🙏👍💥

**Comment by DarrenHuangTW** [comment]
![image](https://github.com/user-attachments/assets/53e11398-3c24-42a0-8c78-472d71ab7300)
Same here, can't find the API key for Google PSE

**Comment by justinh-rahb** [comment]
https://developers.google.com/custom-search/v1/introduction

![Screenshot_20240727-134132.png](https://github.com/user-attachments/assets/767c29f4-0042-4384-b089-10505b54ace2)

**Comment by darkBuddha** [comment]
<img width="437" alt="image" src="https://github.com/user-attachments/assets/54eebef1-b688-4346-9370-8af2d35004d9" />

Any idea?
